### PR TITLE
Redesigned Hexagon Models and implemented Normals

### PR DESCRIPTION
**Pull request by NilsAffing**
Normals added and format changed.

### CODE REVIEW

File: client/src/world/chunk_std.frag [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
-out vec3 color;
+in vec3 toLight;
+in vec3 surfaceNormal;
 
-uniform vec2 offset;
+out vec3 color;
 
 void main() {
-    color = x_color;
+     float diffuse = max(0.0, 1 * dot(toLight, surfaceNormal));
```
Comment (LukasKalbertodt) [inline client/src/world/chunk_std.frag]: This is indented incorrectly and the `1 *` is useless.

File: client/src/world/chunk_view.rs [hunk 2]
```diff
@@ -18,36 +17,59 @@ pub struct ChunkView {
...
+        // for i in 0..6 {
+        // let (x, y) = hex_corner(world::HEX_OUTER_RADIUS, i);
+        //
+        // hexagon_vertex_buffer.push_front(Vertex {
+        // position: [x, y, world::PILLAR_STEP_HEIGHT],
+        // });
+        // hexagon_vertex_buffer.push_back(Vertex { position: [x, y, 0.0] });
+        //
+        // }
+        //
```
Comment (LukasKalbertodt) [inline client/src/world/chunk_view.rs]: Do you need this code in near future? If not, please remove the comments

File: client/src/world/chunk_view.rs [hunk 3]
```diff
@@ -18,36 +17,43 @@ pub struct ChunkView {
...
-        }
-
-        // convert to vector
-        let final_buffer: Vec<_> = hexagon_vertex_buffer.into_iter().collect();
 
-
-        let vbuf = VertexBuffer::new(facade, &final_buffer).unwrap();
+        let mut vertices = Vec::new();
+        let mut indices = Vec::new();
+        let (mut top_vertices, mut top_indices) = get_top_hexagon_modell();
```
Comment (jonas-schievink) [inline client/src/world/chunk_view.rs]: Replace `modell` with `model`

File: client/src/world/chunk_view.rs [hunk 4]
```diff
@@ -153,3 +152,87 @@ impl PillarView {
...
+
+/// Calculates one Point-coordinates of a Hexagon
+fn hex_corner(size: f32, i: i32) -> (f32, f32) {
+    let angle_deg = 60.0 * (i as f32) + 30.0;
+    let angle_rad = (consts::PI / 180.0) * angle_deg;
+
+    (size * angle_rad.cos(), size * angle_rad.sin())
+}
+/// Calculates the top face of the Hexagon and normals
+pub fn get_top_hexagon_modell() -> (Vec<Vertex>, Vec<u32>) {
```
Comment (jonas-schievink) [inline client/src/world/chunk_view.rs]: Do these functions need to be `pub`?

**Pull request closed without merging**